lms/djangoapps/learner_dashboard/api/v0/views.py

```diff
@@ -372,13 +372,15 @@ def get(self, request):
         for course_id in course_keys:
             course_data = get_course_data(course_id)
             if course_data:
-                recommended_courses.append({
-                    'course_run_key': [run['key'] for run in course_data['course_runs']
-                                       if run['availability'] == 'Current'][-1],
-                    'title': str(course_data['title']),
-                    'logo_image_url': course_data['owners'][0]['logo_image_url'],
-                    'marketing_url': course_data.get('marketing_url')
-                })
+                course_run_keys = [course_run['key'] for course_run in course_data['course_runs']
+                                   if course_run['availability'] in ['Current', 'Upcoming']]
+                if course_run_keys:
+                    recommended_courses.append({
+                        'course_run_key': course_run_keys[0],
```

Comment:
In this way, it will always pick the upcoming course run, we should check for current course runs first and then fallback to upcoming if there is no current run.